idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -0,0 +1,662 @@
+        # Ang is used to estimate coeffcients a, b, c for first stage
+        if self.config.first_stage is True:
+            self.Ang = Var(self.flowsheet().config.time,
+                           initialize=0,
+                           doc="Inlet Guide Vanes Angle",
+                           bounds=(-15, 90), units=pyunits.dimensionless)
+
+            @self.Expression(self.flowsheet().config.time)
+            def coeff_a(b, t):
+                return -38.46 * exp(-0.009058 * b.Ang[t]) + (
+                    - 0.1043) * exp(0.09821 * b.Ang[t])
+
+            @self.Expression(self.flowsheet().config.time)
+            def coeff_b(b, t):
+                return 6.863 * exp(-0.01329 * b.Ang[t]) \
+                    + 0.02018 * exp(0.09443 * b.Ang[t])
+
+            @self.Expression(self.flowsheet().config.time)
+            def coeff_c(b, t):
+                return -0.00891 * exp(0.06757 * b.Ang[t]) \
+                    + 0.9661 * exp(0.006228 * b.Ang[t])
+        else:
+            self.coeff_a = Param(default=-428.64,
+                                 mutable=True,
+                                 doc="Performance Curve Parameter A")
+            self.coeff_b = Param(default=29.178,
+                                 mutable=True,
+                                 doc="Perforamance Curve Parameter B")
+            self.coeff_c = Param(default=0.9737,
+                                 mutable=True,
+                                 doc="Performance Curve parameter C")
+
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Correlation between psi_s and psi_3")
+        def psi_s_stage_eqn(b, t):
+            if self.config.first_stage is True:
+                return b.psi_s[t] == b.coeff_a[t] * b.psi_3[t]**2 + \
+                    b.coeff_b[t] * b.psi_3[t] + b.coeff_c[t]
+            else:
+                return b.psi_s[t] == b.coeff_a * b.psi_3[t]**2 + \
+                    b.coeff_b * b.psi_3[t] + b.coeff_c
+
+    def initialize(self, state_args={}, outlvl=idaeslog.NOTSET, solver='ipopt',
+                   optarg={'tol': 1e-6, 'max_iter': 500}):
+        """
+        Initialize the inlet compressor stage model.
+        This deactivates the specialized constraints,
+        then does the isentropic compressor initialization,
+        then reactivates the constraints and solves.
+
+        Args:
+            state_args (dict): Initial state for property initialization
+            outlvl (int): Amount of output (0 to 3) 0 is lowest
+            solver (str): Solver to use for initialization
+            optarg (dict): Solver arguments dictionary
+        """
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="unit")
+
+        opt = SolverFactory(solver)
+        opt.options = optarg
+
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(self, tee=slc.tee)
```

Comment:
Why is the very first thing you are doing a solver call? If this actually works, then there is no point in the rest of this routine as the model is already solved - otherwise this is just going to fail and be a waste of time (or worse, raise a terminal exception).

The first thing you should be doing is calling the initialization routines of the sub-blocks (either control volume or the actual state blocks) and passing in a set of initial guesses. This can also be used to fix the inlet states for you as well, using the `hold_state=True` argument.